From 6f2f415bced22ff908d5270e0728714cf922fd9e Mon Sep 17 00:00:00 2001
From: Nicolas Lalevee <hibou@apache.org>
Date: Tue, 5 Oct 2010 13:16:01 +0000
Subject: [PATCH] IVY-1227: don't use the CacheResolver when useCacheOnly=true
 but rely on the real resolvers, so we hit the proper caches.

git-svn-id: https://svn.apache.org/repos/asf/ant/ivy/core/trunk@1004630 13f79535-47bb-0310-9956-ffa450edef68
---
 src/java/org/apache/ivy/core/resolve/ResolveEngine.java     | 6 +++---
 src/java/org/apache/ivy/plugins/resolver/BasicResolver.java | 6 +++++-
 2 files changed, 8 insertions(+), 4 deletions(-)

diff --git a/src/java/org/apache/ivy/core/resolve/ResolveEngine.java b/src/java/org/apache/ivy/core/resolve/ResolveEngine.java
index f5aef9f52..987e09dda 100644
--- a/src/java/org/apache/ivy/core/resolve/ResolveEngine.java
+++ b/src/java/org/apache/ivy/core/resolve/ResolveEngine.java
@@ -202,9 +202,9 @@ public ResolveReport resolve(ModuleDescriptor md, ResolveOptions options)
             throws ParseException, IOException {
         DependencyResolver oldDictator = getDictatorResolver();
         IvyContext context = IvyContext.getContext();
-        if (options.isUseCacheOnly()) {
-            setDictatorResolver(new CacheResolver(settings));
-        }
+//        if (options.isUseCacheOnly()) {
+//            setDictatorResolver(new CacheResolver(settings));
+//        }
         try {
             String[] confs = options.getConfs(md);
             options.setConfs(confs);
diff --git a/src/java/org/apache/ivy/plugins/resolver/BasicResolver.java b/src/java/org/apache/ivy/plugins/resolver/BasicResolver.java
index c1ef607b0..4763e6386 100644
--- a/src/java/org/apache/ivy/plugins/resolver/BasicResolver.java
+++ b/src/java/org/apache/ivy/plugins/resolver/BasicResolver.java
@@ -217,7 +217,11 @@ public ResolvedModuleRevision getDependency(DependencyDescriptor dd, ResolveData
                     return checkLatest(systemDd, checkForcedResolvedModuleRevision(rmr), data);
                 }
             }
-            
+            if (data.getOptions().isUseCacheOnly()) {
+                throw new UnresolvedDependencyException(
+                    "\t" + getName() + " (useCacheOnly) : no ivy file found for " + systemMrid, false);
+            }
+
             checkInterrupted();
             
             ResolvedResource ivyRef = findIvyFileRef(nsDd, data);
